next-docs/pages/services/indexer.mdx

```diff
@@ -1 +1,168 @@
+import WarningBox from "../../components/WarningBox";
+
 # Indexer
+
+The MUD Indexer is an off-chain indexer for on-chain applications built with MUD.
+
+## Why an off-chain indexer?
+
+Reads with on-chain apps can be tricky.
+What does it mean to be able to query the Ethereum network?
+Technically, given a node with a fully synced state, we can explore just about everything using the EVM, but the “exploring” would be looking at raw storage slots for accounts corresponding to smart contracts.
+A way around this exists by providing view functions on smart contracts: these effectively are just wrappers around raw storage and expose a more friendly API.
+Instead of having to figure out where the balances for an account are stored in the storage tree, we now can call a function that does the lookup via Solidity via an RPC endpoint.
+
+The issue with view functions is that complexity in having to generate and call these to get the “full picture” of the state from the chain explodes pretty quickly.
+For a web app client wanting to present the user with an up-to-date view of the on-chain state, the task of calling the different view functions becomes unnecessarily hard very fast.
+This can also quickly accelerate the need to run a set of dedicated nodes just to be able to service the needed number of RPC requests.
+
+## Installation
```

Comment:
Should we add a paragraph after this that is something like "the MUD indexer solves this problem by listening to the MUD store events to automatically replicate the entire onchain state in a relational database, so it can be queried and served effectively to clients" (feel free to add details and rephrase). Just feel like the intro should include a paragraph about the solution, not just the problem